components/apimgt/org.wso2.carbon.apimgt.api/src/main/java/org/wso2/carbon/apimgt/api/model/APIProduct.java

```diff
@@ -0,0 +1,293 @@
+/*
+*  Copyright (c) 2005-2011, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package org.wso2.carbon.apimgt.api.model;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public class APIProduct {
+    private APIProductIdentifier id;
+    private String uuid;
+
+    private String description;
+    private String productTiers;
+    private Date createdTime;
+    private String createdUser;
+    private Date updatedTime;
+    private String updatedUser;
+    private APIProductStatus status;
+    private String thumbnailUrl;
+    private String apiProductLevelPolicy;
+    private Set<Tier> availableTiers = new LinkedHashSet<Tier>();
+    private String monetizationCategory;
+    private boolean isLatest;
+    private Set<String> tags = new LinkedHashSet<String>();
+
+    private String visibility;
+    private String visibleRoles;
+    private String visibleTenants;
+
+    private String technicalOwner;
+    private String technicalOwnerEmail;
+    private String businessOwner;
+    private String businessOwnerEmail;
+
+    private String apiProductOwner;
+
+    private String subscriptionAvailability;
+    private String subscriptionAvailableTenants;
+
+    public APIProduct(APIProductIdentifier id) {
+        this.id = id;
+    }
+
+    public String getApiProductId() {
```

Comment:
In general, better to explain the purpose of getters setters as Javadoc comment and it makes others life easier when reading the code.